http/server.go

```diff
@@ -658,3 +652,67 @@ func (s *Server) getQuotesPegOutHandler(w http.ResponseWriter, r *http.Request)
 		return
 	}
 }
+
+func returnQuoteSignFunc(w http.ResponseWriter, signature string, depositAddr string) {
+	enc := json.NewEncoder(w)
+	response := acceptRes{
+		Signature:                 signature,
+		BitcoinDepositAddressHash: depositAddr,
+	}
+
+	err := enc.Encode(response)
+	if err != nil {
+		log.Error("error encoding response: ", err.Error())
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+	}
+}
+
+func (s *Server) acceptQuotePegOutHandler(w http.ResponseWriter, r *http.Request) {
+	req := acceptReqPegout{}
+	w.Header().Set("Content-type", "application/json")
+	dec := json.NewDecoder(r.Body)
+	err := dec.Decode(&req)
+
+	if err != nil {
+		log.Error("error decoding request: ", err.Error())
+		http.Error(w, "bad request", http.StatusBadRequest)
+		return
+	}
+
+	quote, err := s.db.GetQuote(req.QuoteHash)
+
+	if err != nil {
+		log.Error("error decoding request: ", err.Error())
+		http.Error(w, "bad request", http.StatusBadRequest)
+		return
+	}
+
+	if quote == nil {
+		log.Error("quote not found:", err.Error())
+		http.Error(w, "not found", http.StatusNotFound)
+		return
+	}
+
+	p := getProviderByAddress(s.providers, quote.LPRSKAddr)
+
+	quoteHashInBytes, err := hex.DecodeString(req.QuoteHash)
+
+	signB, err := p.SignQuote(quoteHashInBytes, req.DerivationAddress, new(types.Wei))
+
+	if err != nil {
+		log.Error("error signing quote: ", err.Error())
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	err = s.addAddressWatcher(quote, req.QuoteHash, req.DerivationAddress, signB, p, types.RQStateWaitingForDeposit)
+	if err != nil {
+		log.Error("error adding address watcher: ", err.Error())
+		http.Error(w, "internal server error", http.StatusInternalServerError)
+		return
+	}
+
+	signature := hex.EncodeToString(signB)
+	returnQuoteSignFunc(w, signature, req.DerivationAddress)
```

Comment:
seems unnecessary and risky to send the derivation address back from the request.